index.html

```diff
@@ -1224,358 +1349,227 @@
-          and abort these steps.
-          </li>
-          <li>If the <em>buffer</em> has been <code><a href=
-          "http://www.w3.org/TR/html51/infrastructure.html#concept-transferable-neutered">
-            neutered</a></code>, Return a promise rejected with an
-            InvalidStateError exception and abort these steps.
-          </li>
-          <li>Create a new <code>ImageBitmap</code> object.
-          </li>
-          <li>Let the <code>ImageBitmap</code> object's bitmap data be the
-          image data given by the <code>BufferSource</code> object,
-          <code><a href=
-          "http://www.w3.org/TR/html51/webappapis.html#cropped-to-the-source-rectangle">
-            cropped to the source rectangle</a></code>.
-          </li>
-          <li>Return a new <code>Promise</code>, but continue running these
-          steps <code><a href=
-          "http://www.w3.org/TR/html51/infrastructure.html#in-parallel">in
-          parallel</a></code>.
-          </li>
-          <li>
-            <code><a href=
-            "http://www.w3.org/TR/html51/infrastructure.html#concept-resolver-fulfill">
-            Resolve</a></code> the <code>Promise</code> with the new
-            <code>ImageBitmap</code> object as the value.
-          </li>
-        </ol>
+        <pre class="idl">
+          [NoInterfaceObject, Exposed=(Window,Worker)]
+          partial interface ImageBitmapFactories {
+              Promise&lt;ImageBitmap&gt; createImageBitmap (BufferSource buffer, long offset, long length, ImageFormat format, ImageFormatPixelLayout layout);
+              Promise&lt;ImageBitmap&gt; createImageBitmap (BufferSource buffer, long offset, long length, ImageFormat format, ImageFormatPixelLayout layout, long sx, long sy, long sw, long sh);
+          };
+        </pre>
+        <div dfn-for="ImageBitmapFactories">
+          <p>
+            The <code><dfn>createImageBitmap</dfn>(buffer, offset, length,
+            format, layout, sx, sy, sw, sh)</code> method must run the
+            following steps:
+          </p>
+          <ol>
+            <li>Let <var>buffer</var> be the container for the raw image data.
+            </li>
+            <li>Let <var>offset</var> be the <a>offset</a> (beginning position)
+            of <var>buffer</var>.
+            </li>
+            <li>Let <var>length</var> be the length of spaces in
+            <var>buffer</var> in which the raw image data is placed into.
+            </li>
+            <li>Let <var>format</var> be the the <a>image format</a> of the raw
+            image data placed in <var>buffer</var>.
+            </li>
+            <li>Let <var>layout</var> be the <a>pixel layout</a> for the raw
+            image data, which describes how the data is arranged in
+            <var>buffer</var> using <var>format</var>.
+            </li>
+            <!-- TODO: do we want to special-case sx, sy handling if undefined? -->
```

Comment:
I am not really sure about this, however, I find that other similar APIs (CanvasRenderingContext::getImageData, CanvasRenderingContext::drawImage) do not specify the "undefined value" special case. Also, in the Firefox implementation, the "undefined value" is regarded as "0" once it is passed as an integer argument.